File: Ahorn/entities/WindCloud.jl
module AnonhelperAnonCloud

using ..Ahorn, Maple

@mapdef Entity "Anonhelper/WindCloud" WindCloud(x::Integer, y::Integer, small::Bool=false, fragile::Bool=false)


const placements = Ahorn.PlacementDict(
    "WindCloud (Anonhelper)" => Ahorn.EntityPlacement(
	WindCloud,
	"point"
   )
)

const sprites = Dict{Tuple{Bool, Bool}, String}(
	(false, false) => "objects/Clouds/windcloud00",
	(true, false) => "objects/Clouds/windfragile00",
	(false, true) => "objects/Clouds/windcloudRemix00",
	(true, true) => "objects/Clouds/windfragileRemix00",
)

function Ahorn.selection(entity::WindCloud)
    fragile = get(entity.data, "fragile", false)
    small = get(entity.data, "small", false)
    sprite = sprites[(fragile, small)]
    x, y = Ahorn.position(entity)
    return Ahorn.getSpriteRectangle(sprite, x, y)
end

function Ahorn.render(ctx::Ahorn.Cairo.CairoContext, entity::WindCloud, room::Maple.Room)
	fragile = get(entity.data, "fragile", false)
	small = get(entity.data, "small", false)
	sprite = sprites[(fragile, small)]
   Ahorn.drawSprite(ctx, sprite, 0, 0)
end

end

File: Loenn/entities/wind_cloud.lua
local utils = require("utils")

local windCloud = {}

windCloud.name = "Anonhelper/WindCloud"
windCloud.depth = 0
windCloud.placements = {
    {
        name = "normal",
        data = {
            fragile = false,
            small = false
        }
    },
    {
        name = "fragile",
        data = {
            fragile = true,
            small = false
        }
    }
}

function windCloud.texture(room, entity)
    local small = entity.small
    local fragile = entity.fragile

    if small then
        if fragile then
            return "objects/AnonHelper/clouds/windfragileRemix00"
        else
            return "objects/AnonHelper/clouds/windcloudRemix00"
        end
    else
        if fragile then
            return "objects/AnonHelper/clouds/windfragile00"
        else
            return "objects/AnonHelper/clouds/windcloud00"
        end
    end
end

function windCloud.selection(room, entity)
	local x, y = entity.x or 0, entity.y or 0
    if entity.small then
        return utils.rectangle(x - 13, y - 6, 24, 15)
    else
        return utils.rectangle(x - 18, y - 6, 36, 15)
    end
end

return windCloud